docs/developer-docs/smart-contracts/overview/development-cycle.mdx

```diff
@@ -0,0 +1,50 @@
+
+## Overview
+
+ICP runs arbitrary [WebAssembly modules in canisters](/docs/current/developer-docs/smart-contracts/overview/introduction). New canisters are created for each individual WebAssembly module that should be run. When a canister is created, it will be assigned a canister ID. There is no central registry of canisters, meaning developers (or more likely their [tools](/docs/current/developer-docs/developer-tools/dev-tools-overview)) have to keep track of their canister IDs. Once code is deployed to canisters, the canister IDs can be used to [call the functions](/docs/current/developer-docs/smart-contracts/call/overview) of other canisters.
+
+## Cycle consumption 
+
+Canisters consume [cycles](/docs/current/developer-docs/gas-cost) while they're running. Performing computation costs cycles, and so does storing data. As long as a canister is running it has to contain enough cycles to maintain operations. This is another place where developers (or their tools) need to keep track of IDs and/or cycles balances. If a canister runs very low on cycles, it will be frozen, meaning it won't respond to update calls anymore, and if it runs out entirely, it will be deleted.
+
+## Access control
+
+Not everyone is allowed to change a canister. Canisters all have a list of [controllers](/docs/current/developer-docs/smart-contracts/maintain/control) that are allowed to update its [settings](/docs/current/developer-docs/smart-contracts/maintain/settings), [install/upgrade](/docs/current/developer-docs/smart-contracts/maintain/upgrade) the running code or even [delete the canister](/docs/current/developer-docs/smart-contracts/maintain/delete). Developers, automated tools, other canisters or nothing at all can be controllers. Depending on the goals for the canister, different combinations of controllers can make sense. [Learn more about canister access control methods](/docs/current/developer-docs/smart-contracts/maintain/control)
+
+## Canister lifecycle 
+
+![Canister lifecycle](_attachments/canister-lifecycle.svg)
+
+A canister's lifecycle follows this general outline:
+
+- A canister is created by compiling code into Wasm, then deploying the code and state onto ICP. 
+
+- A canister runs on ICP and provides a service or function.
+
+- The canister is maintained by its controllers (developer, DAO, etc) through upgrades or being topped-up with cycles.
+
+- A canister is deleted once it is no longer needed or isn't maintained in such that it runs out of cycles. 
+
+## Installing canister code
+
+While canisters can run arbitrary WebAssembly (WASM) modules, ICP has a few conventions that make it easier to get the most out of its capabilities. For example, the function `canister_init` is the first function that gets called after the code is installed for the first time. To facilitate adhering to those conventions, Canister Development Kits (CDKs) exist for [many different languages](/docs/current/developer-docs/developer-tools/dev-tools-overview).
```

Comment:
this might not be entirely accurate - I believe arbitrary wasm cannot be run, it has to be compiled for ICP. I could be wrong - would be good to fact check with @ulan

Reply:
Yes, we have some constraints so it is not arbitrary Wasm. Maybe we should drop that sentence because we have explained this point many times before (about Wasm).

Reply:
Removed this sentence